source/common/ssl/utility.cc

```diff
@@ -101,5 +105,43 @@ SystemTime Utility::getExpirationTime(const X509& cert) {
   return std::chrono::system_clock::from_time_t(days * 24 * 60 * 60 + seconds);
 }
 
+std::vector<uint8_t> Utility::getSha256Digest(const Buffer::Instance& buffer) {
```

Comment:
@PiotrSikora can you do a review pass on these two functions and their tests from a BoringSSL perspective? Thanks.